services/httpd/handler.go

```diff
@@ -1841,3 +1856,64 @@ func (t *Throttler) Handler(h http.Handler) http.Handler {
 		h.ServeHTTP(w, r)
 	})
 }
+
+type statusFilters []*statusFilter
+
+// Match will return true if the status filters are empty or if at least one filter
+// matches the status code.
+func (a statusFilters) Match(statusCode int) bool {
+	if len(a) == 0 {
+		return true
+	}
+
+	for _, sf := range a {
+		if sf.Match(statusCode) {
+			return true
+		}
+	}
+	return false
+}
+
+// statusFilter will check if an http status code matches a certain pattern.
+type statusFilter struct {
+	base    int
+	divisor int
+}
+
+// Match will check if the status code matches this filter.
+func (sf *statusFilter) Match(statusCode int) bool {
+	return statusCode/sf.divisor == sf.base
+}
+
+// reStatusFilter ensures that the format is digits optionally followed by X values.
+var reStatusFilter = regexp.MustCompile(`^(\d+)([xX]*)$`)
+
+// newStatusFilter will create a new status filter from the string.
+func newStatusFilter(s string) (*statusFilter, error) {
```

Comment:
There should be a couple tests for status filter, like testing that `0xx` and `6xx` and `4x0` are rejected filters, `40x` matches `404` but not `411`.

Reply:
Done. `6xx` won't be rejected though since it matches with the pattern. Do you want me to change the regex to `[1-5]` instead of `\d`?

Reply:
I did this anyway so `6xx` will now be rejected.

Reply:
`[1-5]`  would probably be better, if you don't mind. I've skimmed through the spec, and although I don't see where 0xx or 6xx are disallowed, everything in the registered status codes appears to be [100-599].